server/driver_tidb.go

```diff
@@ -155,6 +156,16 @@ func (ts *TiDBStatement) Reset() {
 	}
 }
 
+// StoreStmtCtx implements PreparedStatement StoreStmtCtx method
+func (ts *TiDBStatement) StoreStmtCtx(stmtCtx *stmtctx.StatementContext) {
+	ts.stmtCtx = stmtCtx
+}
+
+// GetStmtCtx implements PreparedStatement GetStmtCtx method
+func (ts *TiDBStatement) GetStmtCtx() *stmtctx.StatementContext {
+	return ts.stmtCtx
+}
```

Comment:
So StmtCtx should be heap allocated...
Currently it's reused

See session/session.go:ExecuteStmt
executor/executor.go:ResetContextOfStmt
sessionctx/variable/session.go:InitStatementContext

@YangKeao